eis_toolkit/cli.py

```diff
@@ -1179,7 +1460,137 @@ def winsorize_transform_cli(
+    from eis_toolkit.validation.calculate_base_metrics import calculate_base_metrics
+
+    typer.echo("Progress: 10%")
+
+    with rasterio.open(input_raster) as src:
+        raster = src.read()
+        typer.echo("Progress: 20%")
+
+    deposits = gpd.read_file(input_deposits)
+    typer.echo("Progress: 40%")
+
+    negatives = gpd.read_file(input_negatives)
+    typer.echo("Progress: 60%")
+
+    metrics = calculate_base_metrics(
+        raster=raster,
+        deposits=deposits,
+        band=band,
+        negatives=negatives,
+    )
+    typer.echo("Progress: 80%")
+
+    metrics.to_csv(output_vector)
+    typer.echo("Progress: 100%")
+    typer.echo(f"Calculate base metrics complete, writing vector to {output_vector}.")
+
+
+# PLOT CORRELATION MATRIX
+# TODO! This will need some work.
+@app.command()
+def correlation_matrix_cli(
+    input_vector: Annotated[Path, INPUT_FILE_OPTION],
+    annotate: bool,
+    cmap: str,
+    plot_title: str,
+    output_png: Annotated[Path, OUTPUT_FILE_OPTION]
+):
+    """Create a Seaborn heatmap to visualize correlation matrix."""
+    from eis_toolkit.validation.plot_correlation_matrix import plot_correlation_matrix
+
+    typer.echo("Progress: 10%")
+
+    geodataframe = gpd.read_file(input_vector)
+    dataframe = pd.DataFrame(geodataframe.drop(columns="geometry"))
+    typer.echo("Progress: 35%")
+
+    result_ax = plot_correlation_matrix(
+        matrix=dataframe,
+        annotate=annotate,
+        cmap=cmap,
+        plot_title=plot_title,
+        kwargs=None)
+    typer.echo("Progress: 70%")
+
+    result_ax.figure.savefig(output_png, format='png', bbox_inches='tight')
+    typer.echo("Progress: 100%")
+    typer.echo(f"Plot correlation matrix complete, writing image to {output_png}.")
+
+
+# TODO Covariance matrix.
+
+
+# PLOT PREDICTION AREA CURVE
+@app.command()
+def plot_prediction_area_curve(
+    input_positives: Annotated[Path, INPUT_FILE_OPTION],
+    proportion_of_area_values: Annotated[Path, INPUT_FILE_OPTION],
+    threshold_values: Annotated[Path, INPUT_FILE_OPTION],
+    out_img: Annotated[Path, OUTPUT_FILE_OPTION]
+):
+    """Plot prediction-area (P-A) plot."""
+    from eis_toolkit.validation.plot_prediction_area_curves import plot_prediction_area_curves
+    typer.echo("Progress: 10%")
+
+    # READ inputs to pd.series. Should this also try to read them into numpy arrays?
+    positives = pd.read_csv(input_positives)
+    typer.echo("Progress: 20%")
+    proportion_of_area = pd.read_csv(proportion_of_area_values)
+    typer.echo("Progress: 40%")
+    thresholds = pd.read_csv(threshold_values)
+    typer.echo("Progress: 60%")
+
+    plot = plot_prediction_area_curves(positives, proportion_of_area, thresholds)
+    typer.echo("Progress: 80")
+
+    plot.figure.savefig(out_img, format='png', bbox_inches='tight')
+    typer.echo("Progress: 100%")
+    typer.echo(f"Plot prediction area curve complete, writing image to {out_img}.")
```

Comment:
Not sure about this either yet (same bucket as base metrics and AUC), also again a plot we could produce in QGIS directly.